src/api/app/controllers/webui/user_controller.rb

```diff
@@ -149,11 +149,20 @@ def icon
   end
 
   def register
-    opts = { login:    params[:login],
-             email:    params[:email],
-             realname: params[:realname],
-             password: params[:password],
-             state:    params[:state] }
+    opts = { login:           params[:login],
+             email:           params[:email],
+             realname:        params[:realname],
+             password:        params[:password],
+             password_repeat: params[:password_repeat],
+             state:           params[:state] }
+
+    if params[:password] != params[:password_repeat]
```

Comment:
Please read the documentation I provided in the link. ``confirmation: true`` will create a virtual attribute whose name is the name of the field that has to be confirmed with "_confirmation" appended, so nothing gets stored in db etc etc.